server/server.js

```diff
@@ -1,15 +1,229 @@
+
 const express = require("express");
 const app = express();
-const port = process.env.PORT || 5000;
+const cors = require ("cors");
+app.use(cors());
+const bodyParser = require("body-parser")
+app.use(bodyParser.json())
+const videos = require("../exampleresponse.json"); 
+app.use(express.json());
+const { body, validationResult } = require("express-validator");
+const port = process.env.PORT || 3000;
+const { Pool } = require("pg");
+
+// const db = new Pool({
+//   user: "shadifakhri", 
+//   host: "localhost",
+//   database: "database",
+//   password: "",
+//   port: 5432,
+// });
+
+const db = new Pool({
+  user: "shadi_user",
+  host: "dpg-cjs7s5dv2qks738v6ltg-a.oregon-postgres.render.com",
+  database: "shadi",
+  password: "j7lLUQoGj3mMw7Q7gH66Nz3ygDizGews",
+  port: 5432,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+// app.get("/videos", function (req, res) {
+//   if (videos.length === 0) {
+//     return res.status(404).json({ error: "no videos found" });
+//   }
+//   res.json(videos);
+// });
 
-// GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+// GET "/videos"
+app.get("/videos", async function (req, res) {
+  const result = await db.query("SELECT * FROM videos");
+  if (result.rows.length === 0) {
+    return res.status(404).json({ error: "no videos found" });
```

Comment:
It's pretty common to only 404 if we're looking for one specific thing that couldn't be found, and if you're asking for "all the videos", to just return the empty array and let the client handle that. What you've done here isn't wrong, it's totally valid, just raising that there are alternatives too :) Can you think of some advantages/disadvantages of both approaches?

Reply:
thanks for the good advice. I changed that.